lib/manager/components/deployment/DeploymentTableRow.js

```diff
@@ -0,0 +1,197 @@
+// @flow
+
+import Icon from '@conveyal/woonerf/components/icon'
+import React, {Component} from 'react'
+import {Button, Badge, Glyphicon} from 'react-bootstrap'
+import {connect} from 'react-redux'
+import { Link } from 'react-router'
+import Select from 'react-select'
+
+import * as deploymentActions from '../../actions/deployments'
+import {formatTimestamp, fromNow} from '../../../common/util/date-time'
+import {
+  versionHasExpired,
+  versionHasNotBegun,
+  versionIsPinnedInDeployment
+} from '../../util/version'
+
+import type {
+  Deployment,
+  Feed,
+  Project,
+  SummarizedFeedVersion
+} from '../../../types'
+import type {AppState} from '../../../types/reducers'
+
+function addClassIfFalsy (val, className) {
+  return !val ? ` ${className}` : ''
+}
+
+function warnIfFalsy (val) {
+  return addClassIfFalsy(val, 'warning')
+}
+
+function dangerIfTruthy (val) {
+  return addClassIfFalsy(!val, 'danger')
+}
+
+type RowProps = {
+  deployment: Deployment,
+  feedSource: Feed,
+  project: Project,
+  version: SummarizedFeedVersion
+}
+
+type ConnectedRowProps = RowProps & {
+  deleteFeedVersion: typeof deploymentActions.deleteFeedVersion,
+  feedVersionPinned: boolean,
+  isVersionEditable: boolean,
+  togglePinFeedVersion: typeof deploymentActions.togglePinFeedVersion,
+  updateVersionForFeedSource: typeof deploymentActions.updateVersionForFeedSource,
+  userCanEditDeployment: boolean,
+  userCanRemoveVersion: boolean
+}
+
+class DeploymentTableRow extends Component<ConnectedRowProps> {
+  _numberToOption = n => ({value: n, label: n})
+
+  _onChangeVersion = option => {
+    const {
+      deployment,
+      feedSource,
+      updateVersionForFeedSource
+    } = this.props
+    const versionToAdd = {feedSourceId: feedSource.id, version: option.value}
+    updateVersionForFeedSource(deployment, feedSource, versionToAdd)
+  }
+
+  _togglePinFeedVersion = () => {
+    const {togglePinFeedVersion, deployment, version} = this.props
+    togglePinFeedVersion(deployment, version)
+  }
+
+  _removeVersion = () => {
+    const {deleteFeedVersion, deployment, version} = this.props
+    deleteFeedVersion(deployment, version)
+  }
+
+  render () {
+    const {
+      feedSource,
+      feedVersionPinned,
+      isVersionEditable,
+      userCanEditDeployment,
+      userCanRemoveVersion,
+      version
+    } = this.props
+    const {latestVersionId} = feedSource
+    const {validationResult: result} = version
+    const expired = versionHasExpired(version)
+    const future = versionHasNotBegun(version)
+    return (
+      <tr>
+        <td>
+          <Link to={`/feed/${feedSource.id}`}>{feedSource.name}</Link>
+          {feedVersionPinned && <Icon type='thumb-tack' />}
```

Comment:
"First impression" comment: Would a padlock icon better express the intent of preventing someone accidentally changing the version? (And would a confirmation message on trying to "unlock" help as well?)

Reply:
I thought you were in favor of the thumbtack/pin?! :)

In any case, I think we should stick with the pin icon to stay consistent with the pinned deployment stuff.